src/qthotkey/meson.build

```diff
@@ -1,17 +1,19 @@
+x11_dep = dependency('x11-xcb', required: false)
 if get_option('qt6')
-  qtx11extras_dep = dependency('qt6', version: qt_req, required: false, modules: ['X11Extras'])
+  # QX11Application requires QtGui private headers and Qt >= 6.2
+  qtx11_dep = dependency('qt6', version: '>= 6.2', private_headers: true, required: false, modules: ['Core', 'Gui'])
 else
-  qtx11extras_dep = dependency('qt5', version: qt_req, required: false, modules: ['X11Extras'])
+  qtx11_dep = dependency('qt5', version: qt_req, required: false, modules: ['X11Extras'])
 endif
 
-have_qthotkey = qtx11extras_dep.found()
+have_qthotkey = qt_dep.found() and x11_dep.found() and qtx11_dep.found()
```

Comment:
We require `qt_dep` in the root meson.build file already, you can omit the check here.